src/library/tlv.hpp

```diff
@@ -217,7 +217,9 @@ class Tlv
 
 Error  GetTlvSet(TlvSet &aTlvSet, const ByteArray &aBuf, Scope aScope = Scope::kMeshCoP);
 TlvPtr GetTlv(tlv::Type aTlvType, const ByteArray &aBuf, Scope aScope = Scope::kMeshCoP);
-bool   IsDatasetParameter(bool aIsActiveDataset, tlv::Type aTlvType);
+Error  GetTlvListByType(TlvList &aTlvList, const ByteArray &aBuf, tlv::Type aTlvType, Scope aScope = Scope::kMeshCoP);
+
+bool IsDatasetParameter(bool aIsActiveDataset, tlv::Type aTlvType);
```

Comment:
nit: is the newline intentional?
```suggestion
Error  GetTlvListByType(TlvList &aTlvList, const ByteArray &aBuf, tlv::Type aTlvType, Scope aScope = Scope::kMeshCoP);
bool   IsDatasetParameter(bool aIsActiveDataset, tlv::Type aTlvType);
```